docs/analyzers/FixAllProvider.md

```diff
@@ -0,0 +1,109 @@
+
+How is a FixAll occurrences code fix computed?
+==============================================
+
+Following steps are used to compute a FixAll occurrences code fix:
+ - Given a specific instance of a diagnostic, compute the set of code actions that claim to fix the diagnostic.
+ - Select a specific code action from this set. In the Visual Studio IDE, this is done by selecting a specific code action in the light bulb menu.
+ - The equivalence key of the selected code action represents the class of code actions that must be applied as part of a FixAll occurrences code fix.
+ - Given this code action, get the FixAll provider corresponding to the code fixer that registered this code action.
+ - If non-null, then request the FixAll provider for its supported FixAllScopes.
+ - Select a specific `FixAllScope` from this set. In the Visual Studio IDE, this is done by clicking on the scope hyperlink in the preview dialog.
+ - Given the trigger diagnostic(s), the equivalence key of the trigger code action, and the FixAll scope, invoke `FixAllProvider.GetFixAsync` to compute the FixAll occurrences code fix.
+
+Adding FixAll support to your code fixer
+========================================
+
+Follow the below steps to add FixAll support to your code fixer:
+ - Override the `CodeFixProvider.GetFixAllProvider` method and return a non-null instance of a `FixAllProvider`. You may either use our built-in FixAllProvider or implement a custom FixAllProvider. See the following sections in this document for determining the correct approach for your fixer.
+ - Ensure that all the code actions registered by your code fixer have a non-null equivalence key. See the following section to determine how to select an equivalence key.
+
+Selecting an Equivalence key for code actions
+=============================================
+
+Each unique equivalence key for a code fixer defines a unique equivalence class of code actions. Equivalence key of the trigger code action is part of the `FixAllContext` and is used to determine the FixAll occurrences code fix.
+Normally, you can use the **'title'** of the code action as the equivalence key. However, there are cases where you may desire to have different values. Let us take an example to get a better understanding.
+
+Let us consider the [C# SimplifyTypeNamesCodeFixProvider](http://source.roslyn.io/#q=Microsoft.CodeAnalysis.CSharp.CodeFixes.SimplifyTypeNames.SimplifyTypeNamesCodeFixProvider) that registers multiple code actions and also has FixAll support. This code fixer offers fixes to simplify the following expressions:
+ - This expressions of the form 'this.x' to 'x'.
+ - Qualified type names of the form 'A.B' to 'B'.
+ - Member access expressions of the form 'A.M' to 'M'.
+
+This fixer needs the following semantics for the corresponding FixAll occurrences code fixes:
+ - This expression simplification: Fix all should simplify all this expressions, regardless of the member being accessed (this.x, this.y, this.z, etc.).
```

Comment:
Maybe write this as - `this` expression simplification: Reading the English word there is confusing